[View in Colaboratory](https://colab.research.google.com/github/kurikei/google-colab-sandbox/blob/master/website/kogolab.chillout.jp/elearn/hamburger/03_chi_square_test.ipynb)

3.3　カイ２乗検定
http://kogolab.chillout.jp/elearn/hamburger/chap3/sec3.html

In [0]:
import numpy as np
import scipy as sp

In [0]:
obs = np.array([[435, 165],[265, 135]])

In [3]:
# 第二引数は 自由度が1のときにイェイツの修正を行うオプション
chi2, p, dof, expected = sp.stats.chi2_contingency(obs, False)

# カイ二乗値
print(chi2)
# 確率
print(p)
# 自由度
print(dof)
# 期待度数
print(expected)

4.464285714285714
0.03461055751570723
1
[[420. 180.]
 [280. 120.]]


3.4　実践編：ハンバーガーを入れてみる
http://kogolab.chillout.jp/elearn/hamburger/chap3/sec4.html

帰無仮説：ワクワクバーガーとモグモグバーガーのポテト・チキン・ハンバーガーの売れ方に違いはない

対立仮説：ワクワクバーガーとモグモグバーガーのポテト・チキン・ハンバーガーの売れ方に違いがある

In [4]:
obs2 = np.array([[435, 165, 650], [265, 135, 350]])
chi2, p, dof, expected =  sp.stats.chi2_contingency(obs2, False)
# カイ二乗値
print(chi2)
# 確率
print(p)
# 自由度
print(dof)
# 期待度数
print(expected)

9.904761904761903
0.007066563731461754
2
[[437.5 187.5 625. ]
 [262.5 112.5 375. ]]


確率は0.007と設定した有意水準1%より小さく、帰無仮説は棄却されるので、売れ方に違いはあるといえる

3.9　通過テスト
http://kogolab.chillout.jp/elearn/hamburger/chap3/sec9.html

In [5]:
obs3 = np.array([[24, 18], [6, 18]])

chi2, p, dof, expected =  sp.stats.chi2_contingency(obs3, False)
# カイ二乗値
print(chi2)
# 確率
print(p)
# 自由度
print(dof)
# 期待度数
print(expected)

6.364285714285714
0.011643995832960663
1
[[19.09090909 22.90909091]
 [10.90909091 13.09090909]]


(1) 桜組と桃組で国語と算数の好き嫌いの傾向に差がない

(2) 桜組と桃組で国語と算数の好き嫌いに差がある

(3) 国語・桜19, 国語・桃23, 算数・桜11, 算数・桃13

(4) 6.36

(5) p = 0.011 で 設定した有意水準1%より大きいので、帰無仮説は棄却できない

(6) このデータからは桜組と桃組で国語と算数の好き嫌いの傾向に差があるとはいえない